cli/flox-rust-sdk/src/models/environment/mod.rs

```diff
@@ -122,12 +122,6 @@ pub struct MigrationInfo {
 }
 
 pub trait Environment: Send {
-    /// Build the environment and create a result link as gc-root
-    fn build(&mut self, flox: &Flox) -> Result<(), EnvironmentError>;
-
-    /// Resolve the environment and return the lockfile
-    fn lock(&mut self, flox: &Flox) -> Result<LockedManifest, EnvironmentError>;
-
```

Comment:
question: why did we take these out completely? To avoid locking independently?
I'm still looking for a command that just rebuilds the env, how would we implement that if not with an `Environment::build()`?

Reply:
It isn't really possible to know which combination of lock, build, and link to perform without more context about what operation is being performed and what type of environment is being operated on. E.g. `list` should never actually `lock` for a RemoteEnvironment, but it should for a PathEnvironment. Because only implementations know when to lock, build, and link, only implementations should be able to lock, build, and link.

Reply:
Yeah that makes sense, if exposing these causes trouble when there are more than one behavior for a single implementation or allow unintended operations like locking remote envs.